convert/convert.go

```diff
@@ -0,0 +1,118 @@
+package convert
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/kong/deck/file"
+	"github.com/kong/deck/utils"
+	"github.com/kong/go-kong/kong"
+)
+
+type Format string
+
+const (
+	FormatKongGateway Format = "kong-gateway"
+	FormatKonnect     Format = "konnect"
+)
+
+var (
+	AllFormats = []Format{FormatKongGateway, FormatKonnect}
+
+	conversions = map[Format]map[Format]bool{
+		FormatKongGateway: {
+			FormatKonnect: true,
+		},
+	}
+)
+
+func validConversion(from, to Format) (bool, error) {
+	if fromMap, ok := conversions[from]; ok {
+		if _, ok := fromMap[to]; ok {
+			return ok, nil
+		}
+	}
+
+	return false, fmt.Errorf("cannot convert from '%s' to '%s' format", from, to)
+}
+
+func ParseFormat(key string) (Format, error) {
+	format := Format(strings.ToLower(key))
+	switch format {
+	case FormatKongGateway:
+		return FormatKongGateway, nil
+	case FormatKonnect:
+		return FormatKonnect, nil
+	default:
+		return "", fmt.Errorf("invalid format: '%v'", key)
+	}
+}
+
+func Convert(inputFilename, outputFilename string, from, to Format) error {
+	if valid, err := validConversion(from, to); !valid {
+		return err
+	}
+
+	key := from + to
+	switch key {
+	case FormatKongGateway + FormatKonnect:
+		return convertKongGatewayToKonnect(inputFilename, outputFilename)
+	default:
+		return fmt.Errorf("cannot convert from '%s' to '%s' format", from, to)
+	}
+}
+
+func convertKongGatewayToKonnect(inputFilename, outputFilename string) error {
```

Comment:
I'm unhappy that this function combines the responsibility with format conversion with the responsibility of storage, but I understand that refactoring the `package file` public interface is probably too much for this PR.

_Just a rant, no action required_

Reply:
Fair. `file `package certainly needs a bit of work and should be split into a couple of packages.